resources/windows/wix/appdatafiles.wxs

```diff
@@ -160,6 +156,32 @@
           Directory="PuppetConfDir" />
       </Component>
       <Component
+        Id="PuppetConfHiera"
+        Permanent="yes"
+        NeverOverwrite="yes"
+        Guid="998A7670-BF42-4F03-B144-4528EADD0CE3"
+        Directory="PuppetConfDir">
+        <CreateFolder />
+        <File
+          Id="HieraConf"
+          KeyPath="yes"
+          Source="SourceDir\CommonAppDataFolder\Puppetlabs\puppet\etc\hiera.yaml" />
+        <Condition>NOT PUP_PREEXISTING_CODE_HIERA AND NOT PUP_PREEXISTING_CONF_HIERA</Condition>
```

Comment:
This conditional blocks the installation of this component (file) if either of these properties exist.